python/cudf/cudf/core/dataframe.py

```diff
@@ -3040,6 +3040,28 @@ def rename(
         else:
             return out.copy(deep=copy)
 
+    def add_prefix(self, prefix):
+        """
+        Prefix labels with string 'prefix'.
+        For DataFrame, the column labels are prefixed.
+        """
```

Comment:
> Do we need to specify docstrings for the series / dataframe implementations of `add_prefix`/`add_suffix` or can we let the `IndexedFrame` docstring get inherited? cc @brandon-b-miller

Docstrings get inherited to `Series` & `DataFrame` from `IndexedFrame`

Reply:
What's wrong with just having it in `index_objects.rst`?

Reply:
I have it in 3 files right now: `index_objects.rst`, `series.rst` and `dataframe.rst`, I'm saying this is probably not required and having it in only `index_objects.rst` is fine (as is discussed above). In that case, do I specify any docstring at all for `add_prefix` and `add_suffix` within `Series` and `DataFrame`? 

Thanks for clarifying though 👍 !

Reply:
Taking a step back..Why are we adding an entry of `add_suffix` & `add_prefix` in `Index` documentation(`index_objects.rst`)? Aren't these both methods only available to `Series` & `DataFrame`?

Reply:
@galipremsagar as you said, Docstrings get inherited to `Series` & `DataFrame` from `IndexedFrame`, and the whole idea was to unify the docstrings as in pandas. So I created `add_prefix` and `add_suffix` inside `IndexedFrame` with a `NotImplementedError`, and put the docs inside there. 

The doubts I have are basically -
1. do we have any docstrings for `add_prefix` and `add_suffix` within `Series` and `DataFrame`?
2. which `.rst` file will then contain the entry for these methods?

Reply:
> @galipremsagar as you said, Docstrings get inherited to `Series` & `DataFrame` from `IndexedFrame`, and the whole idea was to unify the docstrings as in pandas. So I created `add_prefix` and `add_suffix` inside `IndexedFrame` with a `NotImplementedError`, and put the docs inside there.

This is correct approach, but since you **also** defined docstrings again in Series/DataFrame methods, those will be picked up as final docstrings rather than the ones in `IndexFrame`. So action item here is to delete docstrings in `Series` & `DataFrame` `add_suffix/prefix` APIs.

> The doubts I have are basically -
> 
> 1. do we have any docstrings for `add_prefix` and `add_suffix` within `Series` and `DataFrame`?
No, we don't need to have docstrings redefined - same reason as above.

> 2. which `.rst` file will then contain the entry for these methods?

`series.rst` & `dataframe.rst`. You will have to delete the entries in `index_objects.rst` as it is incorrect and will result in warnings during the build process.